tests/llmobs/test_llmobs_span_writer.py

```diff
@@ -1,93 +1,76 @@
-import os
 import time
 
 import mock
 import pytest
 
+from ddtrace import config
 from ddtrace.llmobs._writer import LLMObsSpanWriter
 from tests.llmobs._utils import _chat_completion_event
 from tests.llmobs._utils import _completion_event
 
 
-INTAKE_ENDPOINT = "https://llmobs-intake.datad0g.com/api/v2/llmobs"
-DD_SITE = "datad0g.com"
-dd_api_key = os.getenv("DD_API_KEY", default="<not-a-real-api-key>")
+INTAKE_URL = "https://llmobs-intake.%s" % config._dd_site
+INTAKE_ENDPOINT = "%s/api/v2/llmobs" % INTAKE_URL
 
 
 def test_writer_start(mock_writer_logs):
-    llmobs_span_writer = LLMObsSpanWriter(site="datad0g.com", api_key="asdf", interval=1000, timeout=1)
+    llmobs_span_writer = LLMObsSpanWriter(is_agentless=True, interval=1000, timeout=1)
     llmobs_span_writer.start()
-    mock_writer_logs.debug.assert_has_calls([mock.call("started %r to %r", "LLMObsSpanWriter", INTAKE_ENDPOINT)])
+    mock_writer_logs.debug.assert_has_calls([mock.call("started %r to %r", "LLMObsSpanWriter", INTAKE_URL)])
 
 
 def test_buffer_limit(mock_writer_logs):
-    llmobs_span_writer = LLMObsSpanWriter(site="datad0g.com", api_key="asdf", interval=1000, timeout=1)
+    llmobs_span_writer = LLMObsSpanWriter(is_agentless=True, interval=1000, timeout=1)
     for _ in range(1001):
         llmobs_span_writer.enqueue({})
     mock_writer_logs.warning.assert_called_with(
-        "%r event buffer full (limit is %d), dropping event", "LLMObsSpanWriter", 1000
+        "%r event buffer full (limit is %d), dropping event", "LLMObsSpanEncoder", 1000
     )
 
 
 @pytest.mark.vcr_logs
 def test_send_completion_event(mock_writer_logs):
-    llmobs_span_writer = LLMObsSpanWriter(site="datad0g.com", api_key=dd_api_key, interval=1, timeout=1)
+    llmobs_span_writer = LLMObsSpanWriter(is_agentless=True, interval=1, timeout=1)
     llmobs_span_writer.start()
     llmobs_span_writer.enqueue(_completion_event())
     llmobs_span_writer.periodic()
-    mock_writer_logs.debug.assert_has_calls([mock.call("sent %d LLMObs %s events to %s", 1, "span", INTAKE_ENDPOINT)])
+    mock_writer_logs.debug.assert_has_calls([mock.call("encode %d LLMObs span events to be sent", 1)])
 
 
 @pytest.mark.vcr_logs
 def test_send_chat_completion_event(mock_writer_logs):
-    llmobs_span_writer = LLMObsSpanWriter(site="datad0g.com", api_key=dd_api_key, interval=1, timeout=1)
+    llmobs_span_writer = LLMObsSpanWriter(is_agentless=True, interval=1, timeout=1)
     llmobs_span_writer.start()
     llmobs_span_writer.enqueue(_chat_completion_event())
     llmobs_span_writer.periodic()
-    mock_writer_logs.debug.assert_has_calls([mock.call("sent %d LLMObs %s events to %s", 1, "span", INTAKE_ENDPOINT)])
-
-
-@pytest.mark.vcr_logs
-def test_send_completion_bad_api_key(mock_writer_logs):
```

Comment:
Why did we get rid of this test case?

Reply:
similar to the datadog site change, I also have the span writer pull in an API key from config. since we don't pass in an API key, we don't need to test a case where we pass in an API key.

we could test this if we had a way to set the API key config (or add back passing an API key to test this case)

Reply:
If we can use the `override_global_config()` helper, then I think this should be addressed - I'd like to ensure we have testing for both cases (agent proxy and agentless) if possible 😄

Reply:
I think the test failure here is because the previous test relied on the direct request/response from the naive `LLMObsSpanWriter` with the LLMObs public intake. I think `HTTPWriter` handles 403 responses differently (different logs I think)? Might be worth figuring out what the actual logs/error handling are from 403 responses in the HTTPWriter.